recipes/msgpack/all/conanfile.py

```diff
@@ -0,0 +1,77 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class MsgpackConan(ConanFile):
+    name = "msgpack"
+    description = "The official C++ library for MessagePack"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/msgpack/msgpack-c"
+    topics = ("conan", "msgpack", "message-pack", "serialization")
+    license = "BSL-1.0"
+    exports_sources = "CMakeLists.txt"
+    generators = "cmake"
+    settings = "os", "arch", "build_type", "compiler"
+    options = {"fPIC": [True, False], "shared": [True, False], "header_only": [True, False], "with_boost": [True, False]}
+    default_options = {"fPIC": True, "shared": False, "header_only": False, "with_boost": False}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.header_only:
+            self.settings.clear()
+            del self.options.shared
+            del self.options.fPIC
+            del self.options.with_boost
+
+    def requirements(self):
+        if not self.options.header_only and self.options.with_boost:
+            self.requires.add("boost/1.69.0")
```

Comment:
Update to 1.72.0?
The CI might currently choke on that version as it is heaving issues.

Reply:
done, `test_package` seem to compile and run fine with boost 1.72 and `with_boost=True`